src/Utilities/cachingoptimizer.jl

```diff
@@ -80,11 +80,13 @@ typeof(model_cache)}` so it does not support the function
 """
 function CachingOptimizer(
     model_cache::MOI.ModelLike,
-    optimizer::MOI.AbstractOptimizer,
+    optimizer::MOI.AbstractOptimizer;
+    auto_bridge::Bool = false,
 )
     @assert MOI.is_empty(model_cache)
     @assert MOI.is_empty(optimizer)
-    return CachingOptimizer{typeof(optimizer),typeof(model_cache)}(
+    T = auto_bridge ? MOI.AbstractOptimizer : typeof(optimizer)
```

Comment:
JuMP does not call this constructor, it calls the one with the cache and the mode

Reply:
The reasoning behind this is you could use this constructor, but set `auto_bridge=true` to get the behavior. I've removed.